_posts/2018-04-25-productivity-with-scripts-screencapture.md

```diff
@@ -0,0 +1,151 @@
+echo "The path to this screenshot has been copied to your clipboard."
+```
+
+At this point you can also create a file and paste in the above code.
+
+This script may appear complicated at first sight, but it only took under 10 minutes to write!
+
+# Calling our Script
+
+As seen above in the script's comments, it's using an ENV var, `SCREENSHOT_DIR`. If this var is not exported to your shell, the default value in the script will be used.
+Right now my default value is `"$HOME"`. You can either change that default value in your script, or define/export `SCREENSHOT_DIR` in the shell from which you run the script.
+
+As mentioned in the last section, I decided to let this script take 2 arguments.
+Assuming I saved my script to `~/screenshot.sh`, I can invoke it as follows.
+```
+bash ~/screenshot.sh <file_name> [<optional_extension>]
+```
+`<file_name>` - **This is required** By default, the file name is going to be `$SCREENSHOT_DIR/<file_name>.png`
+`<optional_extension>` - (optional) If you don't want the file extension to be `png` you can define it with this arg. This option could be `jpg` for a `.jpg` extension.
+
+All you have to do is:
+1. Run script
+2. Select area to capture with your mouse
+3. The path will be copied to your clipboard! If you need to open the file, you can just type `open`, then paste the path in to your terminal!
+
+# Setting up a bash alias
+The point of this script in the first place was to save time. For that reason you probably don't want to type `bash <path_to_screenshot_script> <file_name> [<optional_extension>]` every single time.
+Luckily, it's very easy to set up a bash alias for this script!
+If you are unfamiliar with `aliases` in bash, there is an excellent overview here http://tldp.org/LDP/abs/html/aliases.html.
+
+I chose to alias the execution of this script to `snap`. So instead of starting this command with `bash <path_to_screenshot_script>`, I can just type `snap`.
+Note that you can make the following alias name whatever you want though...
+
+Add the following to your `~/.bashrc`, `~/.bash_profile`, or anywhere that gets sourced when starting a new shell. (I suggest eventually setting up a file for all `aliases` like a `~/.bash_aliases` file. There are plenty of great tutorials for that online also).
+
+For the purpose of this tutorial we'll put our `alias` in `~/.bash_profile` since this should be getting sourced on your Mac.
+
+```sh
+alias snap="bash ~/scripts/screencapture.sh"
+```
+As long as the above gets sourced when you open your new shell (or run `. ~/.bash_profile` to do in your current terminal), you will be able to use `snap` as an alias for our script!
+
+# Running Script With Our New Alias
+Now that our `alias` is set up, in your terminal, type:
+```
+snap <file_name> [<optional_path>]
+```
+Even though we have an alias, the arguments the script accepts remain the same. Just like we saw when running the script without an alias, you will everything will work exactly as it did before.
```

Comment:
I think there's a typo here: `you will everything will work` to become `everything will work` without the `you will`